utils/githooks/pre-commit

```diff
@@ -36,7 +36,7 @@ targets=(
     '.env'
 )
 
-files=$(git diff --diff-filter=AM --name-only --cached -- "${targets[@]}")
+files=${files:-$(git diff --diff-filter=AM --name-only --cached -- "${targets[@]}")}
```

Comment:
I've needed this before! :)

Reply:
Indeed.  It makes the hook usable outside of git.

Arguably, with how we have structured utils/githooks, we could make a generic tool and then just call it from the hook.